src/main/java/teammates/common/util/Url.java

```diff
@@ -2,14 +2,12 @@
 
 public class Url {
 
+    protected final String baseUrl;
     protected String urlString;
 
-    public Url(String url) {
-        this.urlString = url.replace(getAppUrl(), ""); // force the URL to be relative
-    }
-
-    protected String getAppUrl() {
-        return Config.APP_URL;
+    public Url(String baseUrl, String url) {
```

Comment:
Parameters of this constructor have a redundancy. May be change to `(baseUrl, relativeUrl)`?